conda/notices/core.py

```diff
@@ -124,10 +124,19 @@ def wrapper(*args, **kwargs):
                 logger.error(f"Unable to open cache file: {str(exc)}")
 
             if channel_notice_set is not None:
-                return_value = func(*args, **kwargs)
-                display_notices(channel_notice_set)
-
-                return return_value
+                try:
+                    return_value = func(*args, **kwargs)
+                    display_notices(channel_notice_set)
+
+                    return return_value
+
+                except Exception:
+                    try:
+                        # Remove the notices cache file if we encounter an exception
+                        cache.clear_cache()
```

Comment:
I'm still not sure this is the right approach to use. This means that every time an error is encountered, the cache files get removed and then conda checks again for these notices which is probably not the desired behavior 😭.

Reply:
Any type of error because if any error occurs at all during the command that is decorated with this decorator, we don't want the notices to be shown. At that point, there's so much output on the screen that it would be too much information presented.